src/App.tsx

```diff
@@ -1,204 +1,59 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { TodoCompletedCategory } from './types/TodoCompletedCategory';
+import { filterTodosByComplated } from './utils/filterTodosByCompleted';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedCategory, setCompletedCategory] =
+    useState<TodoCompletedCategory>(TodoCompletedCategory.all);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.noneError);
+  const countOfNotCompletedTodos = todos.filter(todo => !todo.completed).length;
+  const filtredTodos = filterTodosByComplated(todos, completedCategory);
+
+  async function setTodosFromApi() {
+    try {
+      const todosFromApi = await getTodos();
+
+      setTodos(todosFromApi);
+    } catch {
+      setErrorMessage(Errors.loadError);
+    }
   }
 
+  useEffect(() => {
+    setTodosFromApi();
+  }, []);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
-          />
-
-          {/* Add a todo on form submit */}
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+        <TodoHeader isAllTodosCompleted={countOfNotCompletedTodos === 0} />
+        {todos.length !== 0 && (
```

Comment:
use `!!` operator to convert value to a boolean type
```suggestion
        <TodoHeader isAllTodosCompleted={!!countOfNotCompletedTodos} />
        
        {!!todos.length && (
```